mkdocs/docs/HPC/setting_up_python_virtual_environments.md

```diff
@@ -0,0 +1,262 @@
+
+To use the virtual environment, you need to activate it. 
+This will modify the shell environment to use the Python interpreter and packages from the virtual environment.
+
+```bash
+$ source myenv/bin/activate # Activate the virtual environment
+```
+
+### Installing packages in a virtual environment
+
+After activating the virtual environment, you can install packages with `pip`:
+
+```bash
+$ pip install example_package1
+$ pip install example_package2
+```
+
+These packages will be scoped to the virtual environment and will not affect the system-wide Python installation, 
+and are only available when the virtual environment is activated.
+No administrator privileges are required to install packages in a virtual environment.
+
+It is now possible to run Python scripts that use the installed packages in the virtual environment.
+
+!!! note
+    Always prefer to use Python packages that are centrally installed, which are available via the environment modules interface.
+    Modules are compiled and optimized for the HPC cluster, while packages installed with `pip` are not.
+    To check if a package is available as a module, you can use the following command:
+
+    ```bash
+    $ module av package_name
+    ```
+
+    Some Python packages are installed as extensions of modules. For example, `numpy`, `scipy` and `pandas` are 
+    part of the `SciPy-bundle` module. You can use
+
+    ```bash
+    $ module show <module_name>
+    ```
+
+    to check which extensions are included in a module, if any.
+
+
+### Deactivating a virtual environment
+
+When you are done using the virtual environment, you can deactivate it.
+To do that, run:
+
+```bash
+$ deactivate
+```
+
+## Combining virtual environments with centrally installed modules
+
+You can combine Python packages installed in a virtual environment with environment modules. 
+The following script uses PyTorch (which is available as a module)
+and Poutyne (which we assume is not centrally installed):
+
+```python title="pytorch_poutyne.py"
+import torch
+import poutyne
+
+...
+```
+
+We load a PyTorch package as a module and install Poutyne in a virtual environment:
+
+```bash
+$ module load PyTorch/2.1.2-foss-2023a
+$ python -m venv myenv
+$ source myenv/bin/activate
+$ pip install Poutyne
+```
+
+While the virtual environment is activated, we can run the script without any issues:
+
+```bash
+$ python pytorch_poutyne.py
+```
+
+
+## Creating a virtual environment for a specific cluster
+
+To create a virtual environment for a specific cluster, you need to start an interactive shell on that cluster.
+Let's say you want to create a virtual environment on the `donphan` cluster.
+
+```bash
+$ module swap cluster/donphan
+$ qsub -I
+```
+
+After some time, a shell will be started on the `donphan` cluster. 
+You can now create a virtual environment as described in [the first section](#creating-a-venv).
+This virtual environment can be used by jobs running on the `donphan` cluster.
```

Comment:
Maybe already recommend here to use the `$VSC_OS_LOCAL` and `$VSC_ARCH_LOCAL` environment variables to make a cluster-specific virtual environment.

No need to go into details, just mention this:

```shell
python -m venv myenv-${VSC_OS_LOCAL}-{$VSC_ARCH_LOCAL}
source myenv-${VSC_OS_LOCAL}-{$VSC_ARCH_LOCAL}/bin/activate
```

Maybe you can also mention `${VSC_INSTITUTE_CLUSTER}` here, which holds the cluster name.
I doubt many people use all the clusters, they often only use a subset, and the only overlap we have currently is between `donphan` and `joltik`, so this is easier to use than `${VSC_OS_LOCAL}-${VSC_ARCH_LOCAL}` perhaps

Reply:
Fixed in [add note introducing $VSC_INSTITUTE_CLUSTER](https://github.com/hpcugent/vsc_user_docs/pull/639/commits/6bd51a70e7e2a66bb3c8e679f52b3c532194f3b3). In my opinion, the second option fit the text best